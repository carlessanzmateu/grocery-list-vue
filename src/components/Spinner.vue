<template>
  <div class="spinner w-full flex justify-center items-center fixed inset-0 z-50">
    <img class="image-animation" src="/spinner.gif" alt="spinner" />
  </div>
</template>

<script>
export default {
  name: 'Spinner'
}
</script>

<style lang="scss" scoped>
.spinner {
  .image-animation {
    height: 100px;
    width: 100px;
  }
}
</style>
