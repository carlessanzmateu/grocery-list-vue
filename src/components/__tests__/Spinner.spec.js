import { shallowMount } from '@vue/test-utils'
import Spinner from '@/components/Spinner.vue'

describe('Spinner', () => {
  let wrapper
  function createWrapper() {
    return shallowMount(Spinner)
  }

  it('should create', () => {
    wrapper = createWrapper()
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match html', () => {
    wrapper = createWrapper()
    expect(wrapper.element).toMatchSnapshot()
  })
})
