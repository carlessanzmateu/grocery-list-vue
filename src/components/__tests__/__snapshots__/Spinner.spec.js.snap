// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should match html 1`] = `
<div
  class="spinner w-full flex justify-center items-center fixed inset-0 z-50"
>
  <img
    alt="spinner"
    class="image-animation"
    src="/spinner.gif"
  />
</div>
`;
